mediatimestamp/immutable/timestamp.py

```diff
@@ -18,17 +18,38 @@
 from datetime import datetime
 from dateutil import tz
 from fractions import Fraction
-
-from typing import Tuple, Optional, cast
+from typing import Tuple, Optional, cast, TYPE_CHECKING
+from typing_extensions import Protocol, runtime_checkable
 
 from ..constants import UTC_LEAP
 from ..exceptions import TsValueError
 
 from ._parse import _parse_iso8601
 from ._types import RationalTypes
-from .timeoffset import TimeOffset, TimeOffsetConstructionType
+from .timeoffset import TimeOffset, TimeOffsetConstructionType, mediatimeoffset
+
+if TYPE_CHECKING:
+    from .timerange import TimeRange  # noqa: F401
+
+__all__ = ["Timestamp", "SupportsMediaTimestamp", "mediatimestamp"]
+
+
+@runtime_checkable
+class SupportsMediaTimestamp (Protocol):
+    def __mediatimestamp__(self) -> "Timestamp":
```

Comment:
This appears to be different to TimeOffset and TimeRange which place this under `if TYPE_CHECKING` and have the equivalent of `SupportsMediaTimestamp` with a `__subclasshook__`. Is this intentional?

Reply:
Yes, the other cases need more complex runtime checking than the @runtime_checkable macro provides, but this one doesn't.